source/euler/hyperbolic_system.h

```diff
@@ -1313,5 +1339,159 @@ namespace ryujin
+       U.extract_component(pressure, dim+1);//extract density
+
+       //extract momentum, and convert to velocity
+       for(int c=0; c<dim; c++)
+       {
+         int comp = c+1;//momentum is stored in positions [1,...,dim], so add one to c
+         U.extract_component(momentum.at(c), comp);
+       }
+
+       //extract energy
+       U.extract_component(pressure, dim+1);
+
+       //convert E to pressure
+       for (unsigned int k=0; k<offline_data_.n_locally_owned(); k++)
+       {
+
+         //calculate momentum norm squared
+         const double &E = pressure.local_element(k);
+         const double &rho = density.local_element(k);
+         double m_square = 0;
+         for(int d=0; d<dim; d++)
+           m_square += std::pow(momentum.at(d).local_element(k),2);
+
+         //pressure = (gamma-1)*internal_energy
+         pressure.local_element(k) =  (gamma() - 1.0) * (E - 0.5*m_square/rho);
+       }
+
+
+       if((pressure.has_ghost_elements() != true) || (density.has_ghost_elements() != true)
+           ||(momentum.at(0).has_ghost_elements() != true) || (momentum.at(1).has_ghost_elements() != true))
+       {
+         //TODO: make this some type of exception
+         std::cout << "ghost elements not allowed " << std::endl;
+         std::cout << "pressure " << (pressure.has_ghost_elements() != true) << std::endl;
+         std::cout << "momx " << (momentum.at(0).has_ghost_elements() != true) << std::endl;
+         std::cout << "momy " << (momentum.at(1).has_ghost_elements() != true) << std::endl;
+         std::cout << "density " << (density.has_ghost_elements() != true) << std::endl;
+
+         exit(EXIT_FAILURE);//TODO: remove, for now, I want this to just quit the program
+       }
+       density.update_ghost_values();
+       pressure.update_ghost_values();
+       for(auto mom: momentum)
+         mom.update_ghost_values();
+
+
+       for(const auto& cell : offline_data_.dof_handler().active_cell_iterators())
+       {
+         if(cell->is_locally_owned())
+         {
+           for(unsigned int face = 0; face < cell->n_faces(); ++face)
+           {
+             if(cell->face(face)->at_boundary()
+                 && cell->face(face)->boundary_id() == ryujin::Boundary::slip)
+             {
+               //if on circle, we do the calculation
+               //first, find if the face center is on the circle
+               const auto tria_iterator = cell->face(face);//->center(true);//center of the face on the manifold
+               const dealii::Point<dim> face_center = tria_iterator->center();
+               const double distance = face_center.distance(disk_center);
+
+   //            std::cout << "face center " << face_center << std::endl;
+   //            std::cout << "distance " << distance << std::endl;
+               //check if we are on circle
+               if(distance < disk_radius + 1e-6)//TODO: ask W for a better way to decide if a face is on circle
```

Comment:
Perhaps use the boundary_id for this, rather than using the geometric position of an edge midpoint.

For curved boundaries, you also have the problem that even if the vertices are on the curved boundary, the face center may not. I suspect that this code does not do what you want it to. Compare this with how we determine in step-1 where a face is.

Reply:
Also, in general and independent of @bangerth's comments, I would write such a distance check like this:
```
if(std::abs(distance - disk_radius) < 1e-6)
```

Reply:
Good call on these. I have written an additional boundary  id in a more recent version of this project so that I can more easily control which boundary elements I actually touch. The problem with the boundary id now is that the object I actually want to do analysis on (found with the distance calculation) shares a boundary id with another boundary. A quirk of the ryujin project I guess. It would be much better to specify a boundary id for each separate boundary to avoid this.